### PR TITLE
[XCM refactoring] Get Assets

### PR DESCRIPTION
**Pull request by bobo-k2**
**Pull Request Summary**

I decided to do a series of small PRs related do XCM refactoring to make code review process easier and to spot possible issues with architecture as earlier as possible.
I am trying to make impact to existing XCM functionality as lower as possible.
I believe it make sense to use Vuex to store assets and to propagate changes to UI components.

**Check list**
- [ ] contains breaking changes
- [x] adds new feature
- [x] modifies existing feature (bug fix or improvements)
- [ ] relies on other tasks
- [ ] documentation changes
- [ ] tested on mobile devices

**This pull request makes the following changes:**

**Adds**
- get assets functionality
- assets Vuex store

**Changes**
- implemented price repository with failover which gets asset price from first available price API, instead of relaying on single API (ref. https://github.com/AstarNetwork/astar-apps/pull/435#discussion_r893445212).

### CODE REVIEW

**Comment by impelcrypto** [comment]
Great work, @bobo-k2 .
Currently, I'm working on supporting HRMP transfer in [this](https://github.com/AstarNetwork/astar-apps/pull/460) PR. Considering code conflicts and the strict deadline, I would like to merge PR#461 after merging PR#460.
CC: @hoonsubin @sirius651

**Comment by hoonsubin** [comment]
> Great work, @bobo-k2 . Currently, I'm working on supporting HRMP transfer in [this](https://github.com/AstarNetwork/astar-apps/pull/460) PR. Considering code conflicts and the strict deadline, I would like to merge PR#461 after merging PR#460. CC: @hoonsubin @sirius651

I agree with this. I think it is wise to first deliver on the HRMP asset transfers and clean up the code later.

**Comment by bobo-k2** [comment]
> > Great work, @bobo-k2 . Currently, I'm working on supporting HRMP transfer in [this](https://github.com/AstarNetwork/astar-apps/pull/460) PR. Considering code conflicts and the strict deadline, I would like to merge PR#461 after merging PR#460. CC: @hoonsubin @sirius651
> 
> I agree with this. I think it is wise to first deliver on the HRMP asset transfers and clean up the code later.

@impelcrypto thank you for the review. I resolved your comments and do a few fixes in [this](https://github.com/AstarNetwork/astar-apps/pull/461/commits/3611a4c9923ed4d3c01cc45c898bb73e6fa68a9a) commit. I agree that we merge this PR after #460

File: src/components/assets/Assets.vue
Comment (impelcrypto) [inline src/components/assets/Assets.vue:98]: This value returns an empty array in H160 mode.
image

**Review by impelcrypto — CHANGES_REQUESTED** [review]

File: src/components/assets/Assets.vue
Comment (impelcrypto) [reply src/components/assets/Assets.vue:98]: https://github.com/AstarNetwork/astar-apps/blob/90f9100828691f8130db50ee5f181fabf431be0c/src/components/assets/Assets.vue#L105

Could you share with me how to reproduce an infinite loop? It seems like the issue has been resolved after removing this if statement.

image

Comment (impelcrypto) [reply src/components/assets/Assets.vue:98]: @bobo-k2 
Please take a look at [this](https://github.com/AstarNetwork/astar-apps/pull/464) PR.

Comment (bobo-k2) [reply src/components/assets/Assets.vue:98]: > https://github.com/AstarNetwork/astar-apps/blob/90f9100828691f8130db50ee5f181fabf431be0c/src/components/assets/Assets.vue#L105
> 
> Could you share with me how to reproduce an infinite loop? It seems like the issue has been resolved after removing this if statement.
> 

Just remove if statement and switch to Metamask

Comment (bobo-k2) [reply src/components/assets/Assets.vue:98]: > @bobo-k2 It works as expected on my end after merging [this](https://github.com/AstarNetwork/astar-apps/pull/464) PR. Please let me know if you encounter the same issue again.

Thanks @impelcrypto works great

**Review by hoonsubin — APPROVED** [review]

**Pull request merged**
